src/Mod/Sketcher/Gui/DrawSketchHandlerBSplineByInterpolation.h

```diff
@@ -0,0 +1,606 @@
+                renderSuggestConstraintsCursor(sugConstr.back());
+                return;
+            }
+        }
+        else if (Mode==STATUS_SEEK_ADDITIONAL_POINTS) {
+
+            drawControlPolygonToPosition(onSketchPos);
+
+            drawCursorToPosition(onSketchPos);
+
+            if (seekAutoConstraint(sugConstr.back(), onSketchPos, Base::Vector2d(0.f,0.f))) {
+                renderSuggestConstraintsCursor(sugConstr.back());
+                return;
+            }
+        }
+    }
+
+    bool pressButton(Base::Vector2d onSketchPos) override
+    {
+        prevCursorPosition = onSketchPos;
+
+        MousePressMode = MOUSE_PRESSED;
+
+        if (Mode == STATUS_SEEK_FIRST_POINT) {
+            BSplineKnots.push_back(onSketchPos);
+            BSplineMults.push_back(1); // NOTE: not strictly true for end-points
+
+            Mode = STATUS_SEEK_ADDITIONAL_POINTS;
+
+            // insert point for knot, defer internal alignment constraining.
+            try {
+                Gui::Command::openCommand(QT_TRANSLATE_NOOP("Command", "Add Knot Point"));
+
+                //Add knot
+                Gui::cmdAppObjectArgs(sketchgui->getObject(), "addGeometry(Part.Point(App.Vector(%f,%f,0)),True)",
+                                      BSplineKnots.back().x, BSplineKnots.back().y);
+
+                knotGeoIds.push_back(getHighestCurveIndex());
+            }
+            catch (const Base::Exception& e) {
+                Base::Console().Error("%s\n", e.what());
+                Gui::Command::abortCommand();
+
+                static_cast<Sketcher::SketchObject *>(sketchgui->getObject())->solve();
+
+                return false;
+            }
+
+            //Gui::Command::commitCommand();
+
+            //static_cast<Sketcher::SketchObject *>(sketchgui->getObject())->solve();
+
+            // add auto constraints on knot
+            if (!sugConstr.back().empty()) {
+                createAutoConstraints(sugConstr.back(), knotGeoIds.back(), Sketcher::PointPos::start, false);
+            }
+
+            static_cast<Sketcher::SketchObject *>(sketchgui->getObject())->solve();
+
+            addSugConstraint();
+
+        }
+        else if (Mode == STATUS_SEEK_ADDITIONAL_POINTS) {
+            BSplineKnots.push_back(onSketchPos);
+            BSplineMults.push_back(1); // NOTE: not strictly true for end-points
+
+            // check if coincident with first knot
+            for(auto & ac : sugConstr.back()) {
+                if( ac.Type == Sketcher::Coincident && ac.GeoId == knotGeoIds[0] && ac.PosId == Sketcher::PointPos::start ) {
+                    IsClosed = true;
+                }
+            }
+
+            if (IsClosed) {
+                Mode = STATUS_CLOSE;
+
+                if (ConstrMethod == 1) { // if periodic we do not need the last pole
+                    // FIXME: Look at `interpolate` and check if we need to do this for knots. Periodic splines are as of writing not tested.

```

Comment:
Is this FIXME still true?

Reply:
Needs some work to answer for sure. I'll get back over the weekend.

Reply:
removed